synapse/storage/databases/main/registration.py

```diff
@@ -38,6 +40,35 @@
 logger = logging.getLogger(__name__)
 
 
+@attr.s(frozen=True, slots=True)
+class TokenLookupResult:
+    """Result of looking up an access token.
+
+    Attributes:
+        user_id: The user that this token authenticates as
+        is_guest
+        shadow_banned
+        token_id: The ID of the access token looked up
+        device_id: The device associated with the token, if any.
+        valid_until_ms: The timestamp the token expires, if any.
+        token_owner: The "owner" of the token. This is either the same as the
+            user, or a server admin who is logged in as the user.
+    """
+
+    user_id = attr.ib(type=str)
```

Comment:
it feels confusing that `Request`s have an `authenticated_user` and a `target_user`, `Requester` has `authenticated_entity` and `user`, and this has `token_owner` and a `user_id`. I keep forgetting which is which, particularly the ones called just "user".

Reply:
Agreed. I can certainly change the `user_id` to be `target_user`, I'm not sure about using `authenticated_entity` here since I'm not sure we've actually authenticated.

Reply:
Or something, I'll try and make it better

Reply:
Ok, so now `SynapseRequest` takes a `Requester` object and I changed the opentracing tag from `target_user` to `user_id`. This should mean that everything uses `user`/`user_id` to mean the same thing, so the only difference is `token_owner` vs `authenticated_entity`.

I'm not terribly happy with `Requester.user`, but changing it means touching a *lot* of places, which I'm a bit in two minds about doing.